### PR TITLE
IGNITE-20076 Improve networking shutdown implementation

### PR DESCRIPTION
**Pull request by rpuch**
https://issues.apache.org/jira/browse/IGNITE-20076

### CODE REVIEW

File: modules/network/src/main/java/org/apache/ignite/internal/network/recovery/RecoveryServerHandshakeManager.java [hunk 1]
```diff
@@ -177,12 +182,20 @@ public void onMessage(NetworkMessage message) {
         if (message instanceof HandshakeRejectedMessage) {
             HandshakeRejectedMessage msg = (HandshakeRejectedMessage) message;
 
-            LOG.warn("Handshake rejected by client: {}", msg.reason());
+            boolean ignorable = stopping.get() || !msg.critical();
```
Comment (sashapolo) [inline modules/network/src/main/java/org/apache/ignite/internal/network/recovery/RecoveryServerHandshakeManager.java:185]: Why do we check the stopping flag here?

File: modules/network/src/main/java/org/apache/ignite/internal/network/recovery/RecoveryClientHandshakeManager.java [hunk 2]
```diff
@@ -156,12 +161,20 @@ public void onMessage(NetworkMessage message) {
         if (message instanceof HandshakeRejectedMessage) {
             HandshakeRejectedMessage msg = (HandshakeRejectedMessage) message;
 
-            LOG.warn("Handshake rejected by server: {}", msg.reason());
+            boolean ignorable = stopping.get() || !msg.critical();
```
Comment (sashapolo) [inline modules/network/src/main/java/org/apache/ignite/internal/network/recovery/RecoveryClientHandshakeManager.java:164]: Same question

File: modules/network/src/main/java/org/apache/ignite/internal/network/recovery/RecoveryServerHandshakeManager.java (hunk 1 above)
Comment (rpuch) [reply modules/network/src/main/java/org/apache/ignite/internal/network/recovery/RecoveryServerHandshakeManager.java:185]: If a node are stopping, then the situation 'I was not let by another node to do a handshake' is expected and should neither cause anything (serious) be logged nor fail the node

File: modules/network/src/main/java/org/apache/ignite/internal/network/recovery/RecoveryClientHandshakeManager.java (hunk 2 above)
Comment (rpuch) [reply modules/network/src/main/java/org/apache/ignite/internal/network/recovery/RecoveryClientHandshakeManager.java:164]: The explanation is same :)

File: modules/network/src/main/java/org/apache/ignite/internal/network/recovery/RecoveryServerHandshakeManager.java (hunk 1 above)
Comment (sashapolo) [reply modules/network/src/main/java/org/apache/ignite/internal/network/recovery/RecoveryServerHandshakeManager.java:185]: Agree

**Review by sashapolo — APPROVED** [review]

**Pull request merged**